Description: SONAME convention was broken upstream
 upstream quickly fix the behavior in the 1.5 branch. Import patch
Author: Mathieu Malaterre <malat@debian.org>
Forwarded: http://code.google.com/p/openjpeg/source/detail?r=1625
 http://code.google.com/p/openjpeg/source/detail?r=1464
Last-Update: 2012-04-16

Index: openjpeg/CMakeLists.txt
===================================================================
--- openjpeg.orig/CMakeLists.txt	2012-04-16 16:28:07.000000000 +0200
+++ openjpeg/CMakeLists.txt	2012-04-16 16:28:09.000000000 +0200
@@ -34,9 +34,26 @@
   "${OPENJPEG_VERSION_MAJOR}.${OPENJPEG_VERSION_MINOR}.${OPENJPEG_VERSION_BUILD}")
 SET(PACKAGE_VERSION
   "${OPENJPEG_VERSION_MAJOR}.${OPENJPEG_VERSION_MINOR}.${OPENJPEG_VERSION_BUILD}")
+
+# Because autotools does not support X.Y notation for SOVERSION, we have to use
+# two numerorations, one for the openjpeg version and one for openjpeg soversion
+# version | soversion
+#   1.0   |  0
+#   1.1   |  1
+#   1.2   |  2
+#   1.3   |  3
+#   1.4   |  4
+#   1.5   |  5
+#   2.0   |  6
+# above is the recommendation by the OPJ team. If you really need to override this default,
+# you can specify your own OPENJPEG_SOVERSION at cmake configuration time:
+# cmake -DOPENJPEG_SOVERSION:STRING=42 /path/to/openjpeg
+if(NOT OPENJPEG_SOVERSION)
+  SET(OPENJPEG_SOVERSION 5)
+endif(NOT OPENJPEG_SOVERSION)
 SET(OPENJPEG_LIBRARY_PROPERTIES
   VERSION   "${OPENJPEG_VERSION_MAJOR}.${OPENJPEG_VERSION_MINOR}.${OPENJPEG_VERSION_BUILD}"
-  SOVERSION "${OPENJPEG_VERSION_MAJOR}"
+  SOVERSION "${OPENJPEG_SOVERSION}"
 )
 
 # --------------------------------------------------------------------------
@@ -95,7 +112,7 @@
 ENDIF(NOT OPENJPEG_INSTALL_DATA_DIR)
 
 IF(NOT OPENJPEG_INSTALL_INCLUDE_DIR)
-  SET(OPENJPEG_INSTALL_INCLUDE_DIR "include/")
+  SET(OPENJPEG_INSTALL_INCLUDE_DIR "include/${OPENJPEG_INSTALL_SUBDIR}")
 ENDIF(NOT OPENJPEG_INSTALL_INCLUDE_DIR)
 
 IF(NOT OPENJPEG_INSTALL_MAN_DIR)
@@ -107,7 +124,10 @@
 ENDIF(NOT OPENJPEG_INSTALL_DOC_DIR)
 
 IF(NOT OPENJPEG_INSTALL_PACKAGE_DIR)
-  SET(OPENJPEG_INSTALL_PACKAGE_DIR "${OPENJPEG_INSTALL_SHARE_DIR}/${OPENJPEG_INSTALL_SUBDIR}")
+  # We could install *.cmake files in share/ however those files contains
+  # hardcoded path to libraries on a multi-arch system (fedora/debian) those
+  # path will be different (lib/i386-linux-gnu vs lib/x86_64-linux-gnu)
+  SET(OPENJPEG_INSTALL_PACKAGE_DIR "${OPENJPEG_INSTALL_LIB_DIR}/${OPENJPEG_INSTALL_SUBDIR}")
 ENDIF(NOT OPENJPEG_INSTALL_PACKAGE_DIR)
 
 #-----------------------------------------------------------------------------
@@ -143,13 +163,14 @@
 #-----------------------------------------------------------------------------
 # pkgconfig support
 IF(UNIX)
+  # install in lib and not share (see multi-arch note above)
   CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/libopenjpeg1.pc.cmake
     ${CMAKE_CURRENT_BINARY_DIR}/libopenjpeg1.pc @ONLY)
   INSTALL( FILES  ${CMAKE_CURRENT_BINARY_DIR}/libopenjpeg1.pc DESTINATION
-    ${OPENJPEG_INSTALL_SHARE_DIR}/pkgconfig )
+    ${OPENJPEG_INSTALL_LIB_DIR}/pkgconfig )
   INSTALL( CODE "EXECUTE_PROCESS(COMMAND ${CMAKE_COMMAND} -E create_symlink
   \"libopenjpeg1.pc\"
-  \"\$ENV{DESTDIR}${OPENJPEG_INSTALL_LIB_DIR}/pkgconfig/libopenjpeg.pc\")")
+  \"\$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX}/${OPENJPEG_INSTALL_SHARE_DIR}/pkgconfig/libopenjpeg.pc\")")
 ENDIF(UNIX)
 
 #-----------------------------------------------------------------------------
@@ -254,9 +275,9 @@
 
 #-----------------------------------------------------------------------------
 # install CHANGES and LICENSE
-INSTALL(
-  FILES       CHANGES 
-              LICENSE
-  DESTINATION ${OPENJPEG_INSTALL_DOC_DIR})
+IF(EXISTS ${OPENJPEG_SOURCE_DIR}/CHANGES)
+  INSTALL(FILES CHANGES DESTINATION ${OPENJPEG_INSTALL_DOC_DIR})
+ENDIF(EXISTS ${OPENJPEG_SOURCE_DIR}/CHANGES)
+INSTALL(FILES LICENSE DESTINATION ${OPENJPEG_INSTALL_DOC_DIR})
 
 INCLUDE (CMake/OpenJPEGCPack.cmake)
Index: openjpeg/libopenjpeg/CMakeLists.txt
===================================================================
--- openjpeg.orig/libopenjpeg/CMakeLists.txt	2012-04-16 16:30:40.000000000 +0200
+++ openjpeg/libopenjpeg/CMakeLists.txt	2012-04-16 16:30:48.000000000 +0200
@@ -55,7 +55,7 @@
 
 # Install includes files
 INSTALL(FILES openjpeg.h
-  DESTINATION ${OPENJPEG_INSTALL_INCLUDE_DIR}/${OPENJPEG_INSTALL_SUBDIR} COMPONENT Headers
+  DESTINATION ${OPENJPEG_INSTALL_INCLUDE_DIR} COMPONENT Headers
 )
 
 # install man page of the library
